### ISSUE TITLE
Module declaration cannot refer itself

### ISSUE THREAD

**Issue by hydroper** [Microsoft/TypeScript#14044]
**TypeScript Version:**  2.1.1 / nightly (2.2.0-dev.201xxxxx)

**Code**

```ts
export declare module luaparse {
    /* ... */
    export function write(input: string): luaparse;
    /* ... */
}
```

**Expected behavior:** No exceptions or warnings.

**Actual behavior:** 
![image](https://cloud.githubusercontent.com/assets/15965072/22896314/cb50a51c-f206-11e6-803e-18929226b649.png)

Any ideas? Currently I'm not declaring a `tsconfig.json` file.

**Comment by krryan** [comment]
I'm not sure it's the only way to do so, but in our team we have a lot of interface-module pairs sharing a name, so you could do something like this:

```ts
interface luaparse {
    write(input: string): luaparse;
}

export declare module luaparse {
    export function write(input: string): luaparse;
}
```

**Comment by RyanCavanaugh** [comment]
Did you mean to write `typeof luaparse` ? There's no type named `luaparse` in scope.

**Comment by hydroper** [comment]
@krryan It looks a good idea, but I'll need to export interfaces too, and add declaration-comments.
@RyanCavanaugh Actually there's a module declaration. I want to access this declaration in itself, but unfortunately TypeScript is warning me 'luaparse' cannot be found.

**Comment by RyanCavanaugh** [comment]
A module declaration isn't a type declaration. They're different things.

**Comment by hydroper** [comment]
@RyanCavanaugh Hum, but I'm really needing to tell that the write's method returns the module itself. Or am I confusing something?

**Comment by RyanCavanaugh** [comment]
So write `: typeof luaparse`

**Comment by hydroper** [comment]
@RyanCavanaugh Thanks! It worked
